packages/common/src/hooks/chats/useChatBlastAudienceContent.ts

```diff
@@ -0,0 +1,99 @@
+import { useMemo } from 'react'
+
+import { ChatBlast, ChatBlastAudience } from '@audius/sdk'
+
+import {
+  useGetCurrentUser,
+  useGetCurrentUserId,
+  useGetPlaylistById,
+  useGetPurchasersCount,
+  useGetRemixersCount,
+  useGetTrackById
+} from '~/api'
+import { getChat } from '~/store/pages/chat/selectors'
+import { decodeHashId, getChatBlastTitle } from '~/utils'
+
+import { useProxySelector } from '../useProxySelector'
+
+export const useChatBlastAudienceContent = ({ chatId }: { chatId: string }) => {
+  const chat = useProxySelector((state) => getChat(state, chatId), [chatId])
+  const {
+    audience,
+    audience_content_id: audienceContentId,
+    audience_content_type: audienceContentType
+  } = chat as ChatBlast
```

Comment:
this is potential jank. i can't do `if is_blast return null` bc hooks are below, and they depend on the blast-specific fields.
i could pass the whole chat (of type `ChatBlast`) in but i prefer passing in the ID and selecting at the leaf?

Reply:
chatgpt has offered: `= chat?.is_blast ? chat : ({} as ChatBlast)`

Reply:
this would solve my problems. i can pass the chat object in its entirety and guarantee that it will be a ChatBlast. perhaps i am over-eager in selecting at leaf.